packages-exp/auth-exp/src/core/auth/auth_impl.ts

```diff
@@ -103,6 +101,8 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
   tenantId: string | null = null;
   settings: AuthSettings = { appVerificationDisabledForTesting: false };
 
+  _customTokenProvider: { getCustomToken(): Promise<string> } | null = null;
```

Comment:
nit: define a type for this, something like `export interface CustomTokenProvider { getCustomToken(): Promise<string> }` in https://github.com/firebase/firebase-js-sdk/blob/master/packages-exp/auth-exp/src/model/public_types.ts